akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -455,11 +455,14 @@ private[akka] trait RememberingShard { selfType: Shard ⇒
     context.child(name) match {
       case Some(actor) ⇒
         actor.tell(payload, snd)
-
       case None ⇒
-        //Note; we only do this if remembering, otherwise the buffer is an overhead
-        messageBuffers.append(id, msg, snd)
-        processChange(EntityStarted(id))(sendMsgBuffer)
+        if (state.entities.contains(id)) {
+          getEntity(id).tell(payload, snd)
```

Comment:
Is there a risk that `messageBuffers` already holds some messages and this will bypass those and result in that messages are delivered in wrong order?

Reply:
would it be more safe to 
```
if (state.entities.contains(id)) {
  messageBuffers.append(id, msg, snd)
  sendMsgBuffer(EntityStarted(id))
}
```

Reply:
It isn't possible for messageBuffers to hold anything for this entity at this point in the code because deliverTo is only called from within deliverMessage when there isn't a message buffer for this id (see line 312).

Reply:
ok, thanks, could you add a `require` that message buffer is empty here so that it doesn't go unnoticed if we change something here in the future